src/semver/Version.d

```diff
@@ -0,0 +1,139 @@
+/*******************************************************************************
+
+    Utility to simplify working with semantic versioning entities, including
+    parsing and comparison.
+
+*******************************************************************************/
+
+module semver.Version;
+
+/**
+    Describes SemVer aware version instance
+ **/
+struct Version
+{
+    /// major version number
+    int major;
+
+    /// minor version number
+    int minor;
+
+    /// patch version number
+    int patch;
+
+    /// original version string
+    string representation;
+
+    /**
+        Returns:
+            standard string representation of the version
+     **/
+    string toString ( ) const pure
+    {
+        return this.representation;
+    }
+
+    /**
+        Parses version string of form vX.Y.Z
+
+        If string doesn't match the pattern, it is simply stored as
+        as representation without initializing any numeric version fields.
+
+        Params:
+            ver = string to parse
+
+        Returns:
+            matching Version struct
+     **/
+    static Version fromString ( string ver )
+    {
+        import std.exception : enforce;
+        import std.conv;
+        import std.regex;
+
+        static verRegex = regex(r"^v(\d+)\.(\d+)\.(\d+)$", "g");
+
+        Version result;
+        result.representation = ver;
+
+        auto hit = ver.matchFirst(verRegex);
+        if (hit.empty)
+            return result;
+
+        result.major = to!int(hit[1]);
+        result.minor = to!int(hit[2]);
+        result.patch = to!int(hit[3]);
+        return result;
+    }
+
+    unittest
+    {
+        auto ver = Version.fromString("v30.20.10");
+        assert(ver.major == 30);
+        assert(ver.minor == 20);
+        assert(ver.patch == 10);
+        assert(ver.representation == "v30.20.10");
+
+        ver = Version.fromString("gibberish");
```

Comment:
So this is not validated at all? Shouldn't it throw if the string is not in the expected format?